data-transfer-service/src/main/java/com/epam/pipeline/dts/transfer/service/impl/AutonomousSynchronizationService.java

```diff
@@ -0,0 +1,155 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.dts.transfer.service.impl;
+
+import com.epam.pipeline.dts.transfer.model.AutonomousSyncRule;
+import com.epam.pipeline.dts.transfer.model.StorageItem;
+import com.epam.pipeline.dts.transfer.model.StorageType;
+import com.epam.pipeline.dts.transfer.model.TransferTask;
+import com.epam.pipeline.dts.transfer.model.pipeline.PipelineCredentials;
+import com.epam.pipeline.dts.transfer.repository.TaskRepository;
+import com.epam.pipeline.dts.transfer.service.TransferService;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@EnableScheduling
+public class AutonomousSynchronizationService {
+
+    private static final String SCHEMA_DELIMITER = "://";
+
+    private final TaskRepository taskRepository;
+    private final TransferService transferService;
+    private final PipelineCredentials pipeCredentials;
+    private final Set<AutonomousSyncRule> activeSyncRules;
+    private final Map<AutonomousSyncRule, TransferTask> activeTransferTasks;
+
+    @Autowired
+    public AutonomousSynchronizationService(final @Value("${dts.local.pipe.api.url}") String pipeApiUrl,
+                                            final @Value("${dts.local.pipe.api.token}") String pipeApiToken,
+                                            final TransferService transferService,
+                                            final TaskRepository taskRepository) {
+        this.pipeCredentials = new PipelineCredentials(pipeApiUrl, pipeApiToken);
+        this.taskRepository = taskRepository;
+        this.transferService = transferService;
+        this.activeSyncRules = ConcurrentHashMap.newKeySet();
+        this.activeTransferTasks = new HashMap<>();
+    }
+
+    @Scheduled(cron = "${dts.autonomous.sync.cron}")
+    public void synchronizeFilesystem() {
+        processActiveTasks();
+        submitTasksForAwaitingRules();
+    }
```

Comment:
We need to synchronise directories by cron specified in DTS specific preferences. To do so we can use f.e. `CronSequenceGenerator` to calculate next cron time. With the next cron local calculated we can just check if the time has come and only when start new transfers.